tsdb/chunks/head_chunks.go

```diff
@@ -376,7 +376,29 @@ func repairLastChunkFile(files map[int]string) (_ map[int]string, returnErr erro
 	if err != nil {
 		return files, errors.Wrap(err, "file stat during last head chunk file repair")
 	}
-	if info.Size() == 0 {
+
+	remove := info.Size() == 0 // Empty file.
+	if info.Size() >= MagicChunksSize {
+		// Check if magic number is 0.
+		f, err := os.Open(files[lastFile])
```

Comment:
Do we still need the os.Stat call above if we are almost always going to open the file? OK with it as is though.

Reply:
I was not sure how to check for the length with file open, turns out f.Read will return io.EOF with bytes read which should help here.